### PR TITLE
openrgb: 0.3 -> 0.4

### PR DESCRIPTION
**Pull request by SFrijters**
###### Motivation for this change

https://gitlab.com/CalcProgrammer1/OpenRGB/-/releases/release_0.4

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SFrijters** [comment]
Hm, to me it looks like https://gitlab.com/CalcProgrammer1/OpenRGB/-/blob/master/OpenRGB.pro tries to embed a date into the build, is that something we want to patch out?

**Comment by jonringer** [comment]
I'm not entirely sure, if it inherits from SOURCE_DATE_EPOCH, then we don't have to worry about anything

**Comment by SFrijters** [comment]
It's actually visible in the UI and it looks like my local build time.
2020-09-10_234854_881x388
No git version information because nix removes `.git` I guess?

**Comment by SFrijters** [comment]
Made an attempt at patching the project file; seeing 1 Jan 1980 now in my build. This is something that might be accepted upstream as well.

**Comment by jonringer** [comment]
> No git version information because nix removes `.git` I guess?

correct, there's a "lib.cleanSource" function which will run on most fetchers and remove potential sources of impurity (.git isn't stable)

**Comment by SFrijters** [comment]
Upstream pull request opened at https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/152

**Comment by SFrijters** [comment]
Upstream has accepted my MR, so I've updated the 'patch' commit to reflect that.
Or we can not patch it for now and then the next release should fix it.

File: pkgs/applications/misc/openrgb/default.nix [hunk 1]
```diff
@@ -2,18 +2,26 @@
...
   nativeBuildInputs = [ qmake pkg-config ];
   buildInputs = [ libusb1 hidapi ];
 
+  # For reproducible builds
+  # This change has been accepted upstream so this can be removed next release (0.5)
+  # https://gitlab.com/CalcProgrammer1/OpenRGB/-/commit/f1b7b8ba900db58a1119d8d3e21c1c79de5666aa
+  postPatch = ''
+    substituteInPlace OpenRGB.pro \
+    --replace 'date -R' 'date -R -d "@''${SOURCE_DATE_EPOCH:-$(date +%s)}"'
+  '';
```
Comment (jonringer) [inline pkgs/applications/misc/openrgb/default.nix:23]:
```suggestion
  patches = [
    # Make build SOURCE_DATE_EPOCH aware, merged in master
    (fetchpatch {
      url = "https://gitlab.com/CalcProgrammer1/OpenRGB/-/commit/f1b7b8ba900db58a1119d8d3e21c1c79de5666aa.patch";
      sha256 = "17m1hn1kjxfcmd4p3zjhmr5ar9ng0zfbllq78qxrfcq1a0xrkybx";
    })
  ];
```

Comment (jonringer) [reply pkgs/applications/misc/openrgb/default.nix:23]: and please add `fetchpatch` to the inputs

Comment (SFrijters) [reply pkgs/applications/misc/openrgb/default.nix]: Ah, yes, should've thought of this myself, sorry.

**Pull request merged**